### DISCUSSION TITLE
Compliance Italian D.lgs. n. 24/2023

### DISCUSSION THREAD

**Discussion by simonelucarellisoftec** [globaleaks/globaleaks-whistleblowing-software#3507] [General]
### Proposal

Hi developers!

I'd like to know if developments in order to be compliant with the new italian law (D.lgs. n. 24/2023) are scheduled, and, if they are, when you plan to release them.

We are using your platform in several business environments and we're interested in particolar in 2 features mandatory in italian law:

1. notify the user for received reports ("segnalazioni");
2. alert the user for upcoming 90 days overdue date, starting from 7 days after report ("segnalazione") creation date.

Hope to receive good news!
Thank you,

Simone Lucarelli
Softec S.p.A.
www.softecspa.com

### Motivation and context

Terms of law in Italian D.lgs. n. 24/2023

**Comment by gianlucagilardi** [comment]
Hi there!

1. What  do you mean with "notify the use for received reports"? If the "user" is the whistleblower, they get the receipt of the filed report immediately upon filing (it is the 16-digit code displayed upon filing); if the "user" is the recipient, they already receive an email notifying the report upon the submission itself has been completed.
2. The platform does not alert the whistleblower (quite obviously), but it already has a reminder fore recipients set at 90 days after the report has been filed.

**Comment by simonelucarellisoftec** [comment]
Salve Gianluca, se posso scrivo in Italiano con la speranza di riuscire a
spiegarmi meglio.

1. Mi dicono che la legge prevede che il whistleblower riceva notifica (non
so in quale forma) della "ricezione" intesa come "presa in carico" della
sua segnalazione. Anche a me sembra improbabile, anche considerando che
l'indirizzo email potrebbe non esserci, ma mi dicono che questo è richiesto
dalla normativa.
2. Il secondo punto mi è stato definito come segue: "alert per la scadenza
dei 90 gg per la gestione della segnalazione, decorrenti dai sette gg
successivi al ricevimento della stessa", credo però alert rivolto ai
"riceventi".

A questi punti, aggiungo se possibile una domanda più generale: possiamo
dare la piattaforma come conforme alla nuova normativa italiana in oggetto?

Grazie mille della celerità e precisione,
S.L.

Il giorno gio 29 giu 2023 alle ore 15:14 Gianluca Gilardi <
***@***.***> ha scritto:

> Hi there!
>
>    1. What do you mean with "notify the use for received reports"? If the
>    "user" is the whistleblower, (s)he gets the receipt of the filed report
>    immediately upon filing (it is the 16-digit code displayed upon filing); if
>    the "user" is the recipient, (s)he already receives an email notifying the
>    submission upon the submission itself.
>    2. The platform does not alert the whistleblower (quite obviously),
>    but it already has a reminder fore recipients set at 90 days after the
>    report has been filed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/globaleaks/GlobaLeaks/issues/3506#issuecomment-1613164691>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD4RFJ65ZSWKA64GAF2T4R3XNV5UVANCNFSM6AAAAAAZYPVFJY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Simone Lucarelli*
Technology Team Leader

*P*:  +39 0574 5877
*M*: +39 347 3778724
*W*: *www.softecspa.com <https://www.softecspa.com>*



…………………………………………………….

*Softec S.p.A. *– Viale E. Jenner, 53 20159 Milano – Cap. Soc. Euro
2.497.960,00 - Isc. alla CCIAA di Milano n. 01309040473 - R.E.A. n.
1883334 - P.IVA 01309040473

Questo messaggio di posta elettronica contiene informazioni di
carattere confidenziale rivolte esclusivamente al destinatario sopra
indicato. E' vietato l'uso, la diffusione, la distribuzione o
riproduzione da parte di ogni altra persona. Nel caso aveste ricevuto
questo messaggio di posta elettronica per errore, siete pregati di
segnalarlo immediatamente al mittente e distruggere quanto ricevuto
(compresi i file allegati) senza farne copia. Qualsivoglia utilizzo
non autorizzato del contenuto di questo messaggio costituisce
violazione dell'obbligo di non prendere cognizione della
corrispondenza tra altri soggetti, salvo più grave illecito, ed espone
il responsabile alle relative conseguenze.

Confidentially notice. This e-mail transmission may contain legally
privileged and/or confidential information. Please do not read it if
you are not the intended recipient(S). Any use, distribution,
reproduction or disclosure by any other person is strictly prohibited.
If you have received this e-mail in error, please notify the sender
and destroy the original transmission and its attachments without
reading or saving it in any manner.

**Comment by gianlucagilardi** [comment]
Cerchiamo di gettare un po' di luce :)

1. l'art. 5 del D. Lgs 24/23 testualmente dice:
_" 1. Nell'ambito della gestione del canale di  segnalazione  interna,
la persona o l'ufficio interno ovvero il soggetto esterno,  ai  quali
e' affidata la gestione del canale di segnalazione  interna  svolgono
le seguenti attivita': 
    a) **rilasciano alla persona segnalante avviso di ricevimento della
segnalazione entro sette giorni dalla data di ricezione**; [...]"_

In piattaforma l'"avviso di ricevimento della segnalazione" è, per l'appunto, il codice di 16 cifre che viene rilasciato al momento in cui la segnalazione "inviata al server" è "ricevuta dal server" ed acquisita al sistema.

2. Il reminder **non** è un requisito di compliance: la norma prevede che entro 90 giorni debba essere notiziato il segnalante ("  _d) forniscono riscontro alla segnalazione entro  tre  mesi  dalla data dell'avviso di ricevimento_ [che per noi è la data della segnalazione giusto quanto sopra N.d.R.] _o, in mancanza di tale avviso,  entro tre  mesi  dalla  scadenza  del  termine  di   sette   giorni   dalla presentazione della segnalazione;_ ") ma nulla dice la norma su eventuali reminder ai riceventi.  Detto questo, e' possibile impostare in piattaforma un reminder di scadenza.

Piu' in generale, la piattaforma e' al 99% conforme: sono in rilascio per il 15 luglio due funzionalità (messaggistica vocale e cancellazione selettiva del contenuto della segnalazione) che porteranno la compliance al 100%.

**Comment by simonelucarellisoftec** [comment]
Grazie Gianluca, molto chiaro e decisivo, adottiamo senz'altro la lettura
della normativa e i suggerimenti operativi.




Il giorno gio 29 giu 2023 alle ore 17:11 Gianluca Gilardi <
***@***.***> ha scritto:

> Cerchiamo di gettare un po' di luce :)
>
>    1. l'art. 5 del D. Lgs 24/23 testualmente dice:
>
>
>
>
>
> *" 1. Nell'ambito della gestione del canale di segnalazione interna, la
>    persona o l'ufficio interno ovvero il soggetto esterno, ai quali e'
>    affidata la gestione del canale di segnalazione interna svolgono le
>    seguenti attivita': a) rilasciano alla persona segnalante avviso di
>    ricevimento della segnalazione entro sette giorni dalla data di ricezione;
>    [...]"*
>
> In piattaforma l'"avviso di ricevimento della segnalazione" è, per
> l'appunto, il codice di 16 cifre che viene rilasciato al momento in cui la
> segnalazione "inviata al server" è "ricevuta dal server" ed acquisita al
> sistema.
>
>    1. Il reminder *non* è un requisito di compliance: la norma prevede
>    che entro 90 giorni debba essere notiziato il segnalante (" *d)
>    forniscono riscontro alla segnalazione entro tre mesi dalla data
>    dell'avviso di ricevimento* [che per noi è la data della segnalazione
>    giusto quanto sopra N.d.R.] *o, in mancanza di tale avviso, entro tre
>    mesi dalla scadenza del termine di sette giorni dalla presentazione della
>    segnalazione;* ") ma nulla dice la norma su eventuali reminder ai
>    riceventi. Detto questo, e' possibile impostare in piattaforma un reminder
>    di scadenza.
>
> Piu' in generale, la piattaforma e' al 99% conforme: sono in rilascio per
> il 15 luglio due funzionalità (messaggistica vocale e cancellazione
> selettiva del contenuto della segnalazione) che porteranno la compliance al
> 100%.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/globaleaks/GlobaLeaks/issues/3506#issuecomment-1613356565>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD4RFJ3B7PNRD7FNYAQEPW3XNWLJNANCNFSM6AAAAAAZYPVFJY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Simone Lucarelli*
Technology Team Leader

*P*:  +39 0574 5877
*M*: +39 347 3778724
*W*: *www.softecspa.com <https://www.softecspa.com>*



…………………………………………………….

*Softec S.p.A. *– Viale E. Jenner, 53 20159 Milano – Cap. Soc. Euro
2.497.960,00 - Isc. alla CCIAA di Milano n. 01309040473 - R.E.A. n.
1883334 - P.IVA 01309040473

Questo messaggio di posta elettronica contiene informazioni di
carattere confidenziale rivolte esclusivamente al destinatario sopra
indicato. E' vietato l'uso, la diffusione, la distribuzione o
riproduzione da parte di ogni altra persona. Nel caso aveste ricevuto
questo messaggio di posta elettronica per errore, siete pregati di
segnalarlo immediatamente al mittente e distruggere quanto ricevuto
(compresi i file allegati) senza farne copia. Qualsivoglia utilizzo
non autorizzato del contenuto di questo messaggio costituisce
violazione dell'obbligo di non prendere cognizione della
corrispondenza tra altri soggetti, salvo più grave illecito, ed espone
il responsabile alle relative conseguenze.

Confidentially notice. This e-mail transmission may contain legally
privileged and/or confidential information. Please do not read it if
you are not the intended recipient(S). Any use, distribution,
reproduction or disclosure by any other person is strictly prohibited.
If you have received this e-mail in error, please notify the sender
and destroy the original transmission and its attachments without
reading or saving it in any manner.

**Comment by evilaliv3** [comment]
Hello! I confirm this intepretation that is the same used in italy and other european countries by the different users of globaleaks and also implementors of other digital tools.

We consider the regulator wanted to ensure that in front of a postmail or a voice mail the whistleblower is informed by the confirmation of reception and have possibility to verify the status of the report (e.g. know the handling is overdue); Thes aspect are solved within a digital plaform by informing the user that the report was stored correctly and giving them access that enables them to continuously stay up to date with the handling of their report..

**Comment by marcomeneghin** [comment]
Bungiorno,
  non ho trovato traccia di questi rilasci:
"
sono in rilascio per il 15 luglio due funzionalità (messaggistica vocale e cancellazione selettiva del contenuto della segnalazione) che porteranno la compliance al 100%.
"
Mi confermate che a breve avremo queste due funzionalità?

Grazie, M.

**Comment by gianlucagilardi** [comment]
> Bungiorno, non ho trovato traccia di questi rilasci: " sono in rilascio per il 15 luglio due funzionalità (messaggistica vocale e cancellazione selettiva del contenuto della segnalazione) che porteranno la compliance al 100%. " Mi confermate che a breve avremo queste due funzionalità?
> 
> Grazie, M.

Nella 4.12.3 trovi la messaggistica vocale (c'era gia' nella 4.12.0 ma nel fine settimana abbiamo fixato un po' di cose), per quello che riguarda la cancellazione selettiva la stiamo "allineando" alle linee guida ANAC pubblicate poco piu' di una settimana fa: verrà adottata sia la c.d. "cancellazione logica" (su cui preferisco non commentare) e anche la cancellazione permanente, ma gia' che ci siamo tanto vale essere il piu' possibile allineati alle linee guida :)

**Comment by marcomeneghin** [comment]
Ho testato la messaggistica vocale. Non risco ad impostare correttamente la voce, quando inserisco il criterio di validazione, qulasiasi esso sia cercando di salvare il sistema mi ritorna un errore: "Errore interno del Server (Unexpected)".
C'è qualche modo di risolvere o dobbiamo aspettare la prossima release? Grazie.

**Reply by gianlucagilardi** [reply]
In che senso "impostare il criterio di validazione"? Che mi risulti non esistono criteri di **validazione** applicabili alla voce (non e' un indirizzo email, non e' un numero o numero di telefono e la validazione personalizzata e' altrettanto una regexp, per cui per definizione non applicabile/i alla voce)

**Comment by evilaliv3** [comment]
@marcomeneghin: potresti allegarmi il file /var/globaleaks/log/globaleaks.log ?

**Comment by marcomeneghin** [comment]
Forse è proprio questo il problema, configurando la voce mi viene chiesto di compilare il campo Validazione input:

***@***.***

Grazie, M.


Da: Gianluca Gilardi ***@***.***>
Inviato: martedì 25 luglio 2023 18:07
A: globaleaks/GlobaLeaks ***@***.***>
Cc: Marco Meneghin ***@***.***>; Mention ***@***.***>
Oggetto: Re: [globaleaks/GlobaLeaks] Compliance Italian D.lgs. n. 24/2023 (Discussion #3507)


In che senso "impostare il criterio di validazione"? Che mi risulti non esistono criteri di validazione applicabili alla voce (non e' un indirizzo email,. non e' un numero o numero di telefono e la validazione personalizzata e' altrettanto una regexp, per cui per definizione non applicabile/i alla voce)

**Reply by gianlucagilardi** [reply]
Prova a replicare una configurazione come questa (lascia perdere la condizione di attivazione in alto, non ci badare) e dimmi se hai lo stesso errore
![2023-07-25 18_11_54-glnew - Questionari and 31 more pages - Work - Microsoft​ Edge](https://github.com/globaleaks/GlobaLeaks/assets/2210991/5bfcb0c1-8ced-4559-87e3-c689f0660e31)

**Comment by marcomeneghin** [comment]
Ciao, mi sembra sia già impostata così ma quando salvo mi da l’errore:

***@***.***

***@***.***

Grazie, M.

Da: Gianluca Gilardi ***@***.***>
Inviato: martedì 25 luglio 2023 18:14
A: globaleaks/GlobaLeaks ***@***.***>
Cc: Marco Meneghin ***@***.***>; Mention ***@***.***>
Oggetto: Re: [globaleaks/GlobaLeaks] Compliance Italian D.lgs. n. 24/2023 (Discussion #3507)


Prova a replicare una configurazione come questa (lascia perdere la condizione di attivazione in alto, non ci badare) e dimmi se hai lo stesso errore
[2023-07-25 18_11_54-glnew - Questionari and 31 more pages - Work - Microsoft​ Edge]<https://user-images.githubusercontent.com/2210991/255964974-5bfcb0c1-8ced-4559-87e3-c689f0660e31.png>

**Reply by gianlucagilardi** [reply]
e allora mi sa che bisogna seguire la strada del log come suggerito da @evilaliv3  :)

**Comment by marcomeneghin** [comment]
Ciao,
  in allegato il log e di seguito la notifica di errore che riceviamo via mail quando cerchiamo di configurare la voce orale inserendo:

***@***.***


Platform:

Host: whistleblowing.itstrategy.it (3a4e6nfjyy33htukfcicwk7eytcyrb5jxn5d3v3amrveavgizjr7ehqd.onion)

Version: 4.12.5



KeyError Mapping key not found.



Traceback (most recent call last):



  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 250, in inContext

    result = inContext.theWork()



  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>

    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)



  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 122, in callWithContext

    return self.currentContext().callWithContext(ctx, func, *args, **kw)



  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 85, in callWithContext

    return func(*args,**kw)



  File "/usr/lib/python3/dist-packages/globaleaks/orm.py", line 185, in _wrap

    result = function(session, *args, **kwargs)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 351, in update_field

    field = db_update_field(session, tid, field_id, field, language)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 310, in db_update_field

    db_update_fieldattrs(session, field.id, request['attrs'], language)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in db_update_fieldattrs

    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in <listcomp>

    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 128, in db_update_fieldattr

    if attr_dict['type'] == 'localized' and language is not None:



KeyError: 'type'


Grazie del supporto, M.


Da: Giovanni Pellerano ***@***.***>
Inviato: martedì 25 luglio 2023 17:43
A: globaleaks/GlobaLeaks ***@***.***>
Cc: Marco Meneghin ***@***.***>; Mention ***@***.***>
Oggetto: Re: [globaleaks/GlobaLeaks] Compliance Italian D.lgs. n. 24/2023 (Discussion #3507)


@marcomeneghin<https://github.com/marcomeneghin>: potresti allegarmi il file /var/globaleaks/log/globaleaks.log ?

**Comment by marcomeneghin** [comment]
Buongiorno,
  volevo segnalarvi che con l’ultima versione il problema sembra essersi risolto, compare sempre l’errore se durante la configurazione viene avvalorato il campo “Validazione Input” ma nell’ambiente DEV sono riuscito a caricare delle segnalazioni orali.

A mio avviso ci sono solo un paio di cose da sistemare:

  1.  Il gestore della segnalazione dall’elenco delle segnalazioni sembra avere la possibilità di ascoltare la registrazione ma cliccando sull’anteprima del messaggio non si sente nulla, ha invece la possibilità di settare la velocità del Playback ma non capisco se e come funzioni. Invece entrando nella segnalazione orale poi si riesce a sentire l’audio senza problemi.
***@***.***

  1.  Facendo una segnalazione da Smartphone il riquadro anteprima è troppo piccolo e anche cliccando sui tre puntini il menù compare tutto sulla sinistra e non si riesce a vedere nulla, quindi riascoltare il messaggio è difficoltoso (bisogna ruotare lo smartphone). Ho provato a settare la larghezza del campo nelle proprietà del questionario ma il sistema accetta valori fino al 10 e comunque la sostanza non cambia.
***@***.***


  1.  Se il dispositivo ha il microfono disabilitato sembra comunque che il sistema registri ma poi non visualizza l’anteprima e correttamente non consente l’invio della segnalazione. Ho evidenziato la cosa con un ulteriore messaggio prima del campo registrazione. Non so se sia possibile fare qualcosa di più (es. inibire il pulsante registrazione se il microfono non è attivo).


Ringraziandovi ancora per la collaborazione e il supporto saluto cordialmente.

Marco.


Da: Marco Meneghin
Inviato: giovedì 27 luglio 2023 17:03
A: globaleaks/GlobaLeaks ***@***.***>; globaleaks/GlobaLeaks ***@***.***>
Cc: Mention ***@***.***>; Fabio Battistetti ***@***.***>
Oggetto: R: [globaleaks/GlobaLeaks] Compliance Italian D.lgs. n. 24/2023 (Discussion #3507)

Ciao,
  in allegato il log e di seguito la notifica di errore che riceviamo via mail quando cerchiamo di configurare la voce orale inserendo:

***@***.***


Platform:

Host: whistleblowing.itstrategy.it (3a4e6nfjyy33htukfcicwk7eytcyrb5jxn5d3v3amrveavgizjr7ehqd.onion)

Version: 4.12.5



KeyError Mapping key not found.



Traceback (most recent call last):



  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 250, in inContext

    result = inContext.theWork()



  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 266, in <lambda>

    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)



  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 122, in callWithContext

    return self.currentContext().callWithContext(ctx, func, *args, **kw)



  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 85, in callWithContext

    return func(*args,**kw)



  File "/usr/lib/python3/dist-packages/globaleaks/orm.py", line 185, in _wrap

    result = function(session, *args, **kwargs)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 351, in update_field

    field = db_update_field(session, tid, field_id, field, language)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 310, in db_update_field

    db_update_fieldattrs(session, field.id, request['attrs'], language)



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in db_update_fieldattrs

    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in <listcomp>

    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]



  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 128, in db_update_fieldattr

    if attr_dict['type'] == 'localized' and language is not None:



KeyError: 'type'


Grazie del supporto, M.


Da: Giovanni Pellerano ***@***.******@***.***>>
Inviato: martedì 25 luglio 2023 17:43
A: globaleaks/GlobaLeaks ***@***.******@***.***>>
Cc: Marco Meneghin ***@***.******@***.***>>; Mention ***@***.******@***.***>>
Oggetto: Re: [globaleaks/GlobaLeaks] Compliance Italian D.lgs. n. 24/2023 (Discussion #3507)


@marcomeneghin<https://github.com/marcomeneghin>: potresti allegarmi il file /var/globaleaks/log/globaleaks.log ?

**Comment by PTandreag** [comment]
Ciao a tutti,

perdonate l'intrusione avrei la necessità di capire se il problema che avete descritto sia stato risolto tramite update o tramite workaround.
nello specifico attualmente noi stiamo testando la versione 4.14.8 su docker e volevamo provare ad implementare la funzione del messaggio vocale.
Durante la creazione del questionario, se inserisco l'audio, viene richiesto il campo validazione input che a prescindere dalla selezione restituisce errore come già segnalato da voi.
Se durante il salvataggio consulto il log, l'errore che ritrovo è il seguente 

> 2024-03-11 10:55:17+0000 [-] [E] Unhandled exception raised:
> 2024-03-11 10:55:17+0000 [-] [E] KeyError Mapping key not found.\n\nTraceback (most recent call last):\n\n  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 244, in inContext\n    result = inContext.theWork()  # type: ignore[attr-defined]\n             ^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/twisted/python/threadpool.py", line 260, in <lambda>\n    inContext.theWork = lambda: context.call(  # type: ignore[attr-defined]\n                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 117, in callWithContext\n    return self.currentContext().callWithContext(ctx, func, *args, **kw)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/twisted/python/context.py", line 82, in callWithContext\n    return func(*args, **kw)\n           ^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/globaleaks/orm.py", line 185, in _wrap\n    result = function(session, *args, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 351, in update_field\n    field = db_update_field(session, tid, field_id, field, language)\n            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 310, in db_update_field\n    db_update_fieldattrs(session, field.id, request['attrs'], language)\n\n  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in db_update_fieldattrs\n    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]\n                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 152, in <listcomp>\n    attrs_names = [db_update_fieldattr(session, field_id, attr_name, attr, language) for attr_name, attr in field_attrs.items()]\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n  File "/usr/lib/python3/dist-packages/globaleaks/handlers/admin/field.py", line 128, in db_update_fieldattr\n    if attr_dict['type'] == 'localized' and language is not None:\n       ~~~~~~~~~^^^^^^^^\n\nKeyError: 'type'\n


Se lato client provo a caricare una segnalazione, il questionario sembra registrare il messaggio vocale, salvo poi presentarlo come un rettangolo nero con il quale non è possibile interagire in alcun modo (anche se sembra essere cliccabile poiché il mouse cambia icona al passaggio).

La chiusura della segnalazione avviene senza errori per il whistleblower, ma lato ricevente non trovo alcuna segnalazione inserita.

Purtroppo anche se rifaccio il build del docker pullando a nuovo l'immagine per essere sicuro di avere l'ultima versione disponibile la situazione non cambia.
Qualcuno per caso nel mentre ha scoperto qualcosa in più in merito?

Grazie in anticipo a tutti!
